processor/src/utils.rs

```diff
@@ -40,3 +43,91 @@ pub(crate) fn split_u32_into_u16(value: u64) -> (u16, u16) {
 
     (hi, lo)
 }
+
+// BATCH INVERSION
+// ================================================================================================
+
+/// Parallel batch inversion using Montgomery's trick, with zeros left unchanged.
+///
+/// Processes chunks in parallel using rayon, each chunk using Montgomery's trick.
+pub(crate) fn batch_inversion_par(values: &mut [Felt]) {
```

Comment:
Can we re-use [plonky3](https://github.com/Plonky3/Plonky3/blob/bfb90eb5efc935d5ab763f4393b340c85c567fbb/field/src/batch_inverse.rs#L21)

Reply:
Created https://github.com/0xMiden/miden-vm/issues/2485